services/brig/src/Brig/User/API/Search.hs

```diff
@@ -150,23 +147,10 @@ searchLocally searcherId searchTerm maybeMaxResults = do
   let maxResults = maybe 15 (fromIntegral . fromRange) maybeMaxResults
   teamSearchInfo <- mkTeamSearchInfo
 
-  maybeExactHandleMatch <- exactHandleSearch teamSearchInfo
```

Comment:
There is a problem here: whether or not results are returned for users depends on the searchable flag (team users are by default not searchable by prefix search, but only by exact-handle-in-cassandra-search). The changeset here by no longer doing a request to cassandra omits those results entirely. That might at least partially explain the test failures.

The logic is a bit weird and grew as more requirements were tacked on to the product.